base/range.jl

```diff
@@ -907,11 +907,17 @@ end
 
 isassigned(r::AbstractRange, i::Int) = firstindex(r) <= i <= lastindex(r)
 
+getindex(v::AbstractRange, i::Integer) = _getindex(v, i)
+getindex(v::AbstractRange, i::Bool) = throw(ArgumentError("invalid index: $i of type Bool"))
```

Comment:
```suggestion
function getindex(v::AbstractRange, i::Integer)
    if i isa Bool # Not via dispatch to avoid ambiguities
        throw(ArgumentError("invalid index: $i of type Bool"))
    else
        _getindex(v, i)
    end
end
```

Reply:
:( Ugly but I don't see a better way